pkg/internal/history/history.go

```diff
@@ -0,0 +1,261 @@
+	"github.com/goinggo/mapstructure"
+	"github.com/manifoldco/promptui"
+)
+
+//History struct
+type History struct {
+	ConfigPath string
+	Binary     string
+	Items      []string
+}
+
+//Settings History struct
+func Settings(path string, binary string) *History {
+	return &History{
+		ConfigPath: path,   //Config Path
+		Binary:     binary, //executable Binary file
+	}
+}
+
+//SetPath History path
+func SetPath(path string) *History {
+	return &History{
+		ConfigPath: path, //Config Path
+	}
+}
+
+//Write history record
+func (h *History) Write(i interface{}) {
+	f, err := os.OpenFile(h.ConfigPath, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0666)
+	if err != nil {
+		log.Fatalln("History file write error ", h.ConfigPath)
+	}
+	defer f.Close()
+	if _, err := f.WriteString(convert(i)); err != nil {
+		log.Fatalln("History file write error ", h.ConfigPath)
+	}
+}
+
+//Load the history record
+func (h *History) Load() *History {
+	f, err := os.Open(h.ConfigPath)
+	if err != nil {
+		log.Fatalln("cannot open the file", err)
+	}
+	defer f.Close()
+	rd := bufio.NewReader(f)
+
+	var items []string
+
+	for {
+		line, _, err := rd.ReadLine()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			log.Fatalln("Load err:", err)
+		}
+		items = append(items, string(line))
+	}
+	h.Items = items
+	return h
+}
+
+//List History records order by descending
+func (h *History) List() *History {
+	items := reverse(h.Items)
+	if len(items) > 1000 {
+		temp := make([]string, 1000)
+		copy(temp, items[:1000])
+		h.Items = temp
+		return h
+	}
+	h.Items = items
+	return h
+}
+
+//Select one History item from history
+func (h *History) Select() *History {
+	items := factoryMyBoxItem(h.Items)
+	templates := &promptui.SelectTemplates{
+		Label:    "{{ . }}?",
+		Active:   "\U0001F4CC {{ .Cmd | cyan }} ",
+		Inactive: "  {{ .Cmd | cyan }}",
+		Selected: "\U0001F4CC {{ .Cmd | red | cyan }} ",
+		Details: `
+--------- Cluster Info ----------
+{{ "Garden:" | faint }}	{{ .Garden }}
+{{ "Project:" | faint }}	{{ .Project }}
+{{ "Seed:" | faint }}	{{ .Seed }}
+{{ "Control Plane:" | faint }}	{{ .ControlPlane }}
+{{ "Shoot:" | faint }}	{{ .Shoot }}`,
+	}
+	searcher := func(input string, index int) bool {
+		myboxitem := items[index]
+		name := strings.Replace(strings.ToLower(myboxitem.Cmd), " ", "", -1)
+		input = strings.Replace(strings.ToLower(input), " ", "", -1)
+		return strings.Contains(name, input)
+	}
+
+	prompt := promptui.Select{
+		Label:     "Target hisotry",
+		Items:     items,
+		Templates: templates,
+		Size:      10,
+		Searcher:  searcher,
+	}
+
+	i, _, err := prompt.Run()
+
+	if err != nil {
+		log.Fatalln("Prompt failed: \n", err)
+	}
+	h.Items = strings.Fields(h.Items[i])
+	return h
+}
+
+func factoryMyBoxItem(load []string) []MyBoxItem {
+	items := []MyBoxItem{}
+	box := Mybox{items}
+	for _, i := range load {
+		mapInstance := toMap(i)
+		var myBoxItem MyBoxItem
+		if err := mapstructure.Decode(mapInstance, &myBoxItem); err != nil {
+			fmt.Println(err)
+		}
```

Comment:
why so cumbersome? you could directly set the fields for `MyBoxItem` in the `toMap` / `format` function, without the need to create a `mapstructure`, or not?

Reply:
fixed